From a584bdbf592b1079eb51a3f91c34a378eee341cd Mon Sep 17 00:00:00 2001
From: Fabrice Goucem <fabrice.goucem@nxp.com>
Date: Tue, 23 Nov 2021 19:56:55 +0100
Subject: [PATCH 02/27] arm64: dts: imx8mm-evk-harpoon: remove sai5

sai5 is used by Harpoon's RTOS:
* delete it from Linux device tree
* handle sai5 iomux through iomuxc node

Signed-off-by: Fabrice Goucem <fabrice.goucem@nxp.com>
[remove JIRA ID]
Signed-off-by: Stephane Viau <stephane.viau@nxp.com>
(cherry picked from commit f1803bc0753dc6616a1fbf60387736e05bf500be)
Signed-off-by: Marouen Ghodhbane <marouen.ghodhbane@nxp.com>
---
 .../boot/dts/freescale/imx8mm-evk-harpoon.dts  | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8mm-evk-harpoon.dts b/arch/arm64/boot/dts/freescale/imx8mm-evk-harpoon.dts
index 47e16e606332..7b6f678e831c 100644
--- a/arch/arm64/boot/dts/freescale/imx8mm-evk-harpoon.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mm-evk-harpoon.dts
@@ -6,10 +6,22 @@
 #include "imx8mm-evk-root.dts"
 
 &i2c3 {
-       status = "disabled";
+	status = "disabled";
 };
 
 &iomuxc {
-       pinctrl-names = "default";
-       pinctrl-0 = <&pinctrl_i2c3>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c3>, <&pinctrl_sai5>;
+	pinctrl_sai5: sai5grp {
+		fsl,pins = <
+			MX8MM_IOMUXC_SAI5_RXD1_SAI5_TX_SYNC	0xd6
+			MX8MM_IOMUXC_SAI5_RXD2_SAI5_TX_BCLK	0xd6
+			MX8MM_IOMUXC_SAI5_RXD3_SAI5_TX_DATA0	0xd6
+			MX8MM_IOMUXC_SAI5_RXD0_SAI5_RX_DATA0	0xd6
+		>;
+	};
+};
+
+&sai5 {
+	status = "disabled";
 };
-- 
2.34.1

